src/board_builder/test/accuracy/accuracy_test.py

```diff
@@ -0,0 +1,224 @@
+            return T0_to_T1
+
+        theoretical_transforms = {}
+        for (i, j) in combinations(board_definition.keys(), 2):
+            theoretical_transforms[(i, j)] = calculate_relative_transform(
+                np.array(board_definition[i]), np.array(board_definition[j]))
+
+        # Calculate and compare relative transforms for each snapshot
+        total_difference = 0
+        num_comparisons = 0
+        for snapshot in snapshots:
+            for (i, j) in combinations(snapshot.keys(), 2):
+                snapshot_transform = calculate_relative_transform(
+                    np.array(snapshot[i]), np.array(snapshot[j]))
+                difference = compare_transforms(theoretical_transforms[(i, j)], snapshot_transform)
+                total_difference += difference
+                num_comparisons += 1
+
+        average_difference = total_difference / num_comparisons
+
+        return average_difference
+
+    @staticmethod
+    def _calculate_rms_error_of_two_corner_dataset(simulated_board_arrangement, algorithm_board_arrangement):
+        """
+        Calculates the RMS difference between the output board arrangement with the input board arrangement
+        """
+        def get_corners(target_board):
+            return target_board.get_points()
+
+        simulated_data = get_corners(simulated_board_arrangement)
+        algorithm_data = get_corners(algorithm_board_arrangement)
+
+        if len(simulated_data) != len(algorithm_data):
+            raise ValueError("Theoretical and experimental data must have the same length.")
+
+        rms_error = 0.0
+        total_points = 0
+
+        for arr1, arr2 in zip(simulated_data, algorithm_data):
+            if len(arr1) != len(arr2):
+                raise ValueError("Corresponding arrays must have the same length.")
+            diff = np.array(arr1) - np.array(arr2)
+            rms_error += np.mean(diff ** 2)
+            total_points += 1
+
+        rms_error = np.sqrt(rms_error / total_points)
+        return rms_error
+
+    def run_accuracy_tester(self):
+        module = importlib.import_module(f"src.board_builder.test.accuracy.board_definitions.{self._parameters.SCENE_NAME}")
+        board_definition = getattr(module, "BOARD_DEFINITION")
+
+        ### REFERENCE DATA ###
+        for pose in self._parameters.DETECTOR_POSES_IN_WORLD_REFERENCE:
+            self.board_builder.pose_solver.set_intrinsic_parameters(pose.target_id, self._parameters.DETECTOR_INTRINSICS)
+
+        self.board_builder.pose_solver.set_detector_poses(self._parameters.DETECTOR_POSES_IN_WORLD_REFERENCE)
+
+        ### GENERATE SNAPSHOTS OF THE SCENE ###
+        snapshots = generate_virtual_snapshots(board_definition, self._parameters.NUMBER_OF_SNAPSHOTS)
+        average_difference = self._calculate_generate_snapshots_inaccuracy(board_definition, snapshots)
+        print(f"Generated snapshots inaccuracy (average Frobenius norm difference): {average_difference}")
```

Comment:
It was just to validate my implementation of random board generator. If the value wasn't near 0, it would mean there is an error with the algorithm. I removed it since it is of no use now.